README.md

```diff
@@ -55,32 +55,32 @@ These are the cli options exposed to `pytest` by the plugin.
 | `--snapshot-update`      | When supplied updates existing snapshots of any run tests, as well as deleting unused and generating new snapshots.          |
 | `--snapshot-warn-unused` | Syrupy default behaviour is to fail the test session when there any unused snapshots. This instructs the plugin not to fail. |
 
-### Serializers
+### Syrupy Extensions
 
-Syrupy comes with a few built-in serializers for you to choose from. You should also feel free to extend the AbstractSnapshotSerializer if your project has a need not captured by one our built-ins.
+Syrupy comes with a few built-in preset configurations for you to choose from. You should also feel free to extend the `AbstractSyrupyExtension` if your project has a need not captured by one our built-ins.
 
-- **`AmberSnapshotSerializer`**: This is the default serializer which generates `.ambr` files. Serialization of most data types are supported, however non-sortable types such as frozenset are experimental.
-- **`RawSingleSnapshotSerializer`**: Unlike the `AmberSnapshotSerializer`, which groups all tests within a single test file into a singular snapshot file, the Raw Single serializer creates one `.raw` file per test case.
-- **`PNGSnapshotSerializer`**: An extension of the Raw Single serializer, this should be used to produce `.png` files.
-- **`SVGSnapshotSerializer`**: Another extension of Raw Single. This produces `.svg` files from an svg string.
+- **`AmberSnapshotExtension`**: This is the default extension which generates `.ambr` files. Serialization of most data types are supported, however non-sortable types such as frozenset are experimental.
+- **`RawSingleSnapshotExtension`**: Unlike the `AmberSnapshotExtension`, which groups all tests within a single test file into a singular snapshot file, and creates one `.raw` file per test case.
+- **`PNGSnapshotExtension`**: An extension of raw single, this should be used to produce `.png` files.
+- **`SVGSnapshotExtension`**: Another extension of raw single. This produces `.svg` files from an svg string.
 
-### Advanced Usage, Plugin Support
+### Advanced Usage, Extending Syrupy
 
 ```python
 import pytest
 
 @pytest.fixture
 def snapshot_custom(snapshot):
     return snapshot.with_class(
-        serializer_class=CustomSerializerClass,
+        extension_class=CustomExtensionClass,
     )
```

Comment:
@iamogbz  how about:
```
return snapshot.with_extension(cls=CustomExtensionClass)
```

or something of this sort?

Reply:
I do like that, but wouldn't that be confusing? As `cls` name, while not reserved is usually used to denote the object class

Reply:
was mainly thinking about the `with_extension` rename. Not sure what to call the arg if a named kwarg.